[lutron] Errors caused by malformed responses from Caseta hub

I'm not sure if this is one bug or two.  As a note, I have 3 total bridges.  Two are the Caseta and one is RadioRA2.

Upon initial start of OH I get about 6 of the following error:

```
2019-08-09 08:05:04.155 [WARN ] [ron.internal.handler.IPBridgeHandler] - Integer conversion error parsing update: ~OUTPUT,GNET> 10,1,0.00
2019-08-09 08:05:04.168 [ERROR] [ron.internal.handler.IPBridgeHandler] - Error processing update
java.lang.NumberFormatException: null
        at java.math.BigDecimal.<init>(BigDecimal.java:497) ~[?:?]
        at java.math.BigDecimal.<init>(BigDecimal.java:383) ~[?:?]
        at java.math.BigDecimal.<init>(BigDecimal.java:809) ~[?:?]
        at org.openhab.binding.lutron.internal.handler.DimmerHandler.handleUpdate(DimmerHandler.java:111) ~[243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.handler.IPBridgeHandler.parseUpdates(IPBridgeHandler.java:375) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.handler.IPBridgeHandler.access$0(IPBridgeHandler.java:331) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.handler.IPBridgeHandler$1.inputAvailable(IPBridgeHandler.java:108) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.net.TelnetSession.notifyInputAvailable(TelnetSession.java:86) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.net.TelnetSession.readInput(TelnetSession.java:167) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.net.TelnetSession.access$0(TelnetSession.java:155) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.net.TelnetSession$1.telnetInputAvailable(TelnetSession.java:68) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.apache.commons.net.telnet.TelnetClient.notifyInputListener(TelnetClient.java:420) [233:org.apache.commons.net:3.6.0]
        at org.apache.commons.net.telnet.TelnetInputStream.run(TelnetInputStream.java:650) [233:org.apache.commons.net:3.6.0]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

Then about 2 minutes later I get 3 of the following (which I'm assuming is one per hub):
```
2019-08-09 08:07:13.908 [ERROR] [scovery.LutronDeviceDiscoveryService] - Error scanning for devices
com.thoughtworks.xstream.io.StreamException:  : Connection timed out (Connection timed out)
        at com.thoughtworks.xstream.io.xml.StaxDriver.createReader(StaxDriver.java:126) ~[66:org.apache.servicemix.bundles.xstream:1.4.7.1]
        at com.thoughtworks.xstream.XStream.fromXML(XStream.java:1115) ~[66:org.apache.servicemix.bundles.xstream:1.4.7.1]
        at com.thoughtworks.xstream.XStream.fromXML(XStream.java:1062) ~[66:org.apache.servicemix.bundles.xstream:1.4.7.1]
        at org.eclipse.smarthome.config.xml.util.XmlDocumentReader.readFromXML(XmlDocumentReader.java:85) ~[142:org.openhab.core.config.xml:2.5.0.201908052330]
        at org.openhab.binding.lutron.internal.discovery.LutronDeviceDiscoveryService.readDeviceDatabase(LutronDeviceDiscoveryService.java:92) ~[243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.discovery.LutronDeviceDiscoveryService.access$0(LutronDeviceDiscoveryService.java:88) ~[243:org.openhab.binding.lutron:2.5.0.201908062214]
        at org.openhab.binding.lutron.internal.discovery.LutronDeviceDiscoveryService$1.run(LutronDeviceDiscoveryService.java:75) [243:org.openhab.binding.lutron:2.5.0.201908062214]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.net.ConnectException: Connection timed out (Connection timed out)
        at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:?]
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:?]
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[?:?]
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[?:?]
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[?:?]
        at java.net.Socket.connect(Socket.java:589) ~[?:?]
        at java.net.Socket.connect(Socket.java:538) ~[?:?]
        at sun.net.NetworkClient.doConnect(NetworkClient.java:180) ~[?:?]
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:463) ~[?:?]
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:558) ~[?:?]
        at sun.net.www.http.HttpClient.<init>(HttpClient.java:242) ~[?:?]
        at sun.net.www.http.HttpClient.New(HttpClient.java:339) ~[?:?]
        at sun.net.www.http.HttpClient.New(HttpClient.java:357) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1226) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1162) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1056) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:990) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1570) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498) ~[?:?]
        at java.net.URL.openStream(URL.java:1057) ~[?:?]
        at com.thoughtworks.xstream.io.xml.StaxDriver.createReader(StaxDriver.java:109) ~[?:?]
        ... 13 more
```

// Edited by @cweitkamp: Added code fences to improve readability

--- a/bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/IPBridgeHandler.java
+++ b/bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/IPBridgeHandler.java
@@ -50,7 +50,8 @@ import org.slf4j.LoggerFactory;
  *         LutronHandlerFactory
  */
 public class IPBridgeHandler extends BaseBridgeHandler {
-    private static final Pattern STATUS_REGEX = Pattern.compile("~(OUTPUT|DEVICE|SYSTEM|TIMECLOCK|MODE),([^,]+),(.*)");
+    private static final Pattern RESPONSE_REGEX = Pattern
+            .compile("~(OUTPUT|DEVICE|SYSTEM|TIMECLOCK|MODE),([0-9\\.:/]+),([0-9,\\.:/]*)\\Z");
 
     private static final String DB_UPDATE_DATE_FORMAT = "MM/dd/yyyy HH:mm:ss";
 
@@ -325,6 +326,7 @@ public class IPBridgeHandler extends BaseBridgeHandler {
 
     private void parseUpdates() {
         String paramString;
+        String scrubbedLine;
 
         for (String line : this.session.readLines()) {
             if (line.trim().equals("")) {
@@ -339,9 +341,28 @@ public class IPBridgeHandler extends BaseBridgeHandler {
                 this.keepAliveReconnect.cancel(true);
             }
 
-            Matcher matcher = STATUS_REGEX.matcher(line);
+            Matcher matcher = RESPONSE_REGEX.matcher(line);
+            boolean responseMatched = matcher.find();
 
-            if (matcher.find()) {
+            if (!responseMatched) {
+                // In some cases with Caseta a CLI prompt may be embedded within a received response line.
+                if (line.contains("NET>")) {
+                    // Try to remove it and re-attempt the regex match.
+                    scrubbedLine = line.replaceAll("[GQ]NET> ", "");
+                    matcher = RESPONSE_REGEX.matcher(scrubbedLine);
+                    responseMatched = matcher.find();
+                    if (responseMatched) {
+                        line = scrubbedLine;
+                        logger.debug("Cleaned response line: {}", scrubbedLine);
+                    }
+                }
+            }
+
+            if (!responseMatched) {
+                logger.debug("Ignoring message {}", line);
+                continue;
+            } else {
+                // We have a good response message
                 LutronCommandType type = LutronCommandType.valueOf(matcher.group(1));
 
                 if (type == LutronCommandType.SYSTEM) {
@@ -376,8 +397,6 @@ public class IPBridgeHandler extends BaseBridgeHandler {
                 } else {
                     logger.debug("No thing configured for integration ID {}", integrationId);
                 }
-            } else {
-                logger.debug("Ignoring message {}", line);
             }
         }
     }

